vsintegration/src/FSharp.Editor/Hints/InlineParameterNameHints.fs

```diff
@@ -97,9 +107,16 @@ module InlineParameterNameHints =
             if tupleRanges |> (not << Seq.isEmpty) then tupleRanges else curryRanges
             |> Seq.filter (fun range -> argumentLocations |> (not << isNamedArgument range))
 
+        let argumentNames = 
+            ranges
+            |> Seq.map (getSourceTextAtRange sourceText)
+
         parameters
         |> Seq.zip ranges // Seq.zip is important as List.zip requires equal lengths
         |> Seq.where (snd >> doesParameterNameExist)
+        |> Seq.zip argumentNames
+        |> Seq.where (fun (argumentName, (_, parameter)) -> argumentName <> parameter.DisplayName)
+        |> Seq.map snd
         |> Seq.map getParameterHint
```

Comment:
Can `Seq.choose` be used here instead of where + two maps?

Reply:
Yep, updated